build/configs/artik05x/artik05x_download.sh

```diff
@@ -145,11 +145,18 @@ compute_ocd_commands()
 
 download()
 {
-        parts_default=`grep -A 2 'config FLASH_PART_NAME' ${PARTITION_KCONFIG} | sed -n 's/\tdefault "\(.*\)".*/\1/p'`
-        parts2=${CONFIG_FLASH_PART_NAME:=${parts_default}}
-        parts=`echo $parts2 | sed "s/,/ /g"`
+	parts=$1;
 
-        # Make Openocd commands for parts
+	if [[ -n $parts ]] && [[ "$parts" != "all" ]] && [[ "$parts" != "ALL" ]]
+	then
+		echo "##Download $parts"
```

Comment:
The logic of downloading a single binary has not been added in this patch, then why to add the provision before hand? 
we should add this logic along with downloading a single binary logic, because right now it is giving error if you try "make download bl2"

Reply:
How did you test it? Paste the logs here.

Reply:
Here it the test result:

bsvt@ubuntu:~/gitpublic/tizenrtbtheosam/os$ make download bl2
##Download bl2
ocd command to run: flash_write bl2 /home/bsvt/gitpublic/tizenrtbtheosam/build/configs/artik05x/../../../os/../build/configs/artik053/bin/bl2.bin ;
Generating partition map ... Done
Open On-Chip Debugger 0.10.0-dirty (2017-09-02-08:32)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
adapter speed: 2000 kHz
Info : auto-selecting first available session transport "jtag". To override use 'transport select <transport>'.
force hard breakpoints
trst_and_srst separate srst_gates_jtag trst_push_pull srst_push_pull connect_deassert_srst
adapter_nsrst_assert_width: 50
adapter_nsrst_delay: 100
debug_level: -1
target halted in ARM state due to debug-request, current mode: Supervisor
cpsr: 0x80000153 pc: 0x040d10a8
D-Cache: enabled, I-Cache: enabled
Flashing /home/bsvt/gitpublic/tizenrtbtheosam/build/configs/artik05x/../../../os/../build/configs/artik053/bin/bl2.bin to 'BL2' partition at 0x04010000...
target halted in ARM state due to debug-request, current mode: Supervisor
cpsr: 0x80000153 pc: 0x040d10a0
D-Cache: enabled, I-Cache: enabled
148480 bytes written at address 0x04010000
downloaded 148480 bytes in 2.735641s (53.004 KiB/s)
Flash DONE

Reply:
How did you test it? Paste the logs here.

Reply:
I got below error first, but after taking fresh copy it is **solved** now.
That could be due to some previous setting or some hardware issue.
Sorry for the inconvenience.

test@2030021797:~/projects/test/TizenRT/os$ sudo make download bl2
##Download bl2
ocd command to run: flash_write bl2 /home/test/projects/test/TizenRT/build/configs/artik05x/../../../os/../build/configs/artik053/bin/bl2.bin ;
Generating partition map ... Done
Open On-Chip Debugger 0.10.0-dirty (2017-09-02-08:32)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
adapter speed: 2000 kHz
Info : auto-selecting first available session transport "jtag". To override use 'transport select <transport>'.
force hard breakpoints
trst_and_srst separate srst_gates_jtag trst_push_pull srst_push_pull connect_deassert_srst
adapter_nsrst_assert_width: 50
adapter_nsrst_delay: 100
debug_level: -1

Makefile.unix:521: recipe for target 'download' failed
make: *** [download] Error 1

Reply:
Check board connection and setup, target connection failed. Nothing to do with this commit.

Reply:
Yes it is solved now.
Thanks